File: base/constants/types.verification.go
package constants

type VerificationType string

const (
	VerificationEmail        VerificationType = "MAIL"
	VerificationPhone        VerificationType = "PHNE"
	VerificationAge          VerificationType = "UAGE"
	VerificationAddress      VerificationType = "ADDR"
	VerificationCreditCard   VerificationType = "CRCD"
	VerificationProfilePhoto VerificationType = "PROF"
	VerificationUsername     VerificationType = "UNAM"
)

var AllVerificationTypes = []VerificationType{
	VerificationEmail,
	VerificationPhone,
	VerificationAge,
	VerificationAddress,
	VerificationCreditCard,
	VerificationProfilePhoto,
	VerificationUsername,
}

File: ui/src/interfaces/types.verification.ts
//||------------------------------------------------------------------------------------------------||
//|| Types
//||------------------------------------------------------------------------------------------------||

export type VerificationTypeKeys =
      | "Email"
      | "Phone"
      | "Age"
      | "Address"
      | "CreditCard"
      | "ProfilePhoto"
      | "Username";

export type VerificationTypeMap = Record<VerificationTypeKeys, string>;

//||------------------------------------------------------------------------------------------------||
//|| Constants
//||------------------------------------------------------------------------------------------------||

export const VerificationTypes: VerificationTypeMap = {
      Email:        "MAIL",
      Phone:        "PHNE",
      Age:          "UAGE",
      Address:      "ADDR",
      CreditCard:   "CRCD",
      ProfilePhoto: "PROF",
      Username:     "UNAM",
};

export type VerificationTypeValues = typeof VerificationTypes[VerificationTypeKeys];


//||------------------------------------------------------------------------------------------------||
//|| Utils
//||------------------------------------------------------------------------------------------------||

export function getVerificationKeyFromCode(code: string): VerificationTypeKeys | undefined {
      const entry = Object.entries(VerificationTypes).find(([_, value]) => value === code);
      return entry ? (entry[0] as VerificationTypeKeys) : undefined;
}

export function getAllVerificationTypes(): string[] {
      return Object.values(VerificationTypes);
}

export function getVerificationTypeMap(): Record<string, string> {
      return { ...VerificationTypes };
}
